### PR TITLE
Added new PoMo rate matrices with selection

### PR DESCRIPTION
**Pull request by mrborges23**
Added new PoMo rate matrices with selection: 
1. fnPoMoKN
2. fnReversiblePoMoKN
3. fnPoMo4N
4. fnReversiblePoMo4N
5. fnPoMo2N
6. fnReversiblePoMo2N
7. fnReversiblePoMoBalance4N
8. fnReversiblePoMoTwo4N
9. fnReversiblePoMoThree4N

### CODE REVIEW

File: src/revlanguage/functions/phylogenetics/ratematrix/Func_revPoMo4N.cpp [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+{
+    
+    static ArgumentRules argumentRules = ArgumentRules();
+    static bool          rules_set = false;
+    
+    if ( !rules_set )
+    {
+        argumentRules.push_back( new ArgumentRule( "N"   , Natural::getClassTypeSpec(), "Number of individuals.", ArgumentRule::BY_CONSTANT_REFERENCE, ArgumentRule::ANY ) );
+        argumentRules.push_back( new ArgumentRule( "pi"  , Simplex::getClassTypeSpec(), "Vector of allele frequencies.", ArgumentRule::BY_CONSTANT_REFERENCE, ArgumentRule::ANY ) );
+        argumentRules.push_back( new ArgumentRule( "rho" , ModelVector<RealPos>::getClassTypeSpec(), "Vector of mutation rates: mu=(mu_a0a1,mu_a1a0,mu_a0a2,...).", ArgumentRule::BY_CONSTANT_REFERENCE, ArgumentRule::ANY ) );
```
Comment (bjoelle) [inline src/revlanguage/functions/phylogenetics/ratematrix/Func_revPoMo4N.cpp:72]: is this argument rho or mu ? is it a vector of mutation rates or exchangeabilities ? (l44)

File: src/revlanguage/functions/phylogenetics/ratematrix/Func_revPoMo2N.cpp [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+{
+    
+    static ArgumentRules argumentRules = ArgumentRules();
+    static bool          rules_set = false;
+    
+    if ( !rules_set )
+    {
+        argumentRules.push_back( new ArgumentRule( "N"   , Natural::getClassTypeSpec(), "Number of individuals.", ArgumentRule::BY_CONSTANT_REFERENCE, ArgumentRule::ANY ) );
+        argumentRules.push_back( new ArgumentRule( "pi"  , Simplex::getClassTypeSpec(), "Vector of allele frequencies: pi=(pi_a0,pi_a1).", ArgumentRule::BY_CONSTANT_REFERENCE, ArgumentRule::ANY ) );
+        argumentRules.push_back( new ArgumentRule( "rho" , ModelVector<RealPos>::getClassTypeSpec(), "Vector of mutation rates: mu=(mu_a0a1,mu_a1a0).", ArgumentRule::BY_CONSTANT_REFERENCE, ArgumentRule::ANY ) );
```
Comment (bjoelle) [inline src/revlanguage/functions/phylogenetics/ratematrix/Func_revPoMo2N.cpp:72]: is this argument rho or mu ? is it a vector of mutation rates or exchangeabilities ? (l44)

File: src/revlanguage/functions/phylogenetics/ratematrix/Func_revPoMoKN.cpp [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+    
+    static ArgumentRules argumentRules = ArgumentRules();
+    static bool          rules_set = false;
+    
+    if ( !rules_set )
+    {
+        argumentRules.push_back( new ArgumentRule( "K"   , Natural::getClassTypeSpec(), "Number of alleles.", ArgumentRule::BY_CONSTANT_REFERENCE, ArgumentRule::ANY ) );
+        argumentRules.push_back( new ArgumentRule( "N"   , Natural::getClassTypeSpec(), "Number of individuals.", ArgumentRule::BY_CONSTANT_REFERENCE, ArgumentRule::ANY ) );
+        argumentRules.push_back( new ArgumentRule( "pi"  , Simplex::getClassTypeSpec(), "Vector of allele frequencies.", ArgumentRule::BY_CONSTANT_REFERENCE, ArgumentRule::ANY ) );
+        argumentRules.push_back( new ArgumentRule( "rho" , ModelVector<RealPos>::getClassTypeSpec(), "Vector of mutation rates: mu=(mu_a0a1,mu_a1a0,mu_a0a2,...).", ArgumentRule::BY_CONSTANT_REFERENCE, ArgumentRule::ANY ) );
```
Comment (bjoelle) [inline src/revlanguage/functions/phylogenetics/ratematrix/Func_revPoMoKN.cpp:74]: is this argument rho or mu ? is it a vector of mutation rates or exchangeabilities ? (l45)

**Review by bjoelle — CHANGES_REQUESTED** [review]

File: src/revlanguage/functions/phylogenetics/ratematrix/Func_revPoMoKN.cpp (hunk 3 above)
Comment (mrborges23) [reply src/revlanguage/functions/phylogenetics/ratematrix/Func_revPoMoKN.cpp]: updated pomo rate matrices argument rules and commited

File: src/revlanguage/functions/phylogenetics/ratematrix/Func_revPoMo2N.cpp (hunk 2 above)
Comment (mrborges23) [reply src/revlanguage/functions/phylogenetics/ratematrix/Func_revPoMo2N.cpp]: updated pomo rate matrices argument rules and commited

File: src/revlanguage/functions/phylogenetics/ratematrix/Func_revPoMo4N.cpp (hunk 1 above)
Comment (mrborges23) [reply src/revlanguage/functions/phylogenetics/ratematrix/Func_revPoMo4N.cpp]: updated pomo rate matrices argument rules and committed

**Review by bjoelle — APPROVED** [review]

**Pull request merged**